src/pybind/mgr/prometheus/module.py

```diff
@@ -604,7 +602,9 @@ def get_mgr_status(self):
             self.metrics['mgr_status'].set(_state, (
                 'mgr.{}'.format(mgr),
             ))
-        always_on_modules = mgr_map['always_on_modules'].get(ceph_release, [])
+        # Get all the always on modules
+        always_on_modules = [x for modules in mgr_map['always_on_modules'].values() for x in modules]
```

Comment:
On my master `mgr_map['always_on_modules']` is:
```
{'nautilus': ['balancer', 'crash', 'devicehealth', 'orchestrator_cli', 'progress', 'rbd_support', 'status', 'volumes'], 'octopus': ['balancer', 'crash', 'devicehealth', 'orchestrator', 'pg_autoscaler', 'progress', 'rbd_support', 'status', 'telemetry', 'volumes'], 'pacific': ['balancer', 'crash', 'devicehealth', 'orchestrator', 'pg_autoscaler', 'progress', 'rbd_support', 'status', 'telemetry', 'volumes'], 'last_failure_osd_epoch': 452, 'active_clients': []}
```
And it rises  `TypeError: 'int' object is not iterable`, due to  `'last_failure_osd_epoch': 452` item. And I think you don't want `active_clients` either? And what about the duplicates in the list?

Reply:
And actually, could you please provide more background why you think just listing the current release modules is not correct?

Reply:
It might be correct/enough, I changed this to avoid a failure when getting `ceph_release` from the mgr host versions. Early on when the mgr daemon is started, the mgr version might not be populated yet and so it fails the metrics collection because it can't find the `ceph_release`.

I wonder if there is a better (stable) way of doing this.

btw: I was testing this primarily on nautilus.

Reply:
OK, I changed this commit to just check if `host_version[1]` is populated before trying to get the `ceph_release` from it. I don't see any better way of doing this at the moment.